### DISCUSSION TITLE
Error : Invalid EVM version requested - Foundry Fundamentals: Section 1 - 27. Zksync Local Deploy

### DISCUSSION THREAD

**Discussion by Alessandro-Cavaliere** [Cyfrin/foundry-full-course-cu#2135] [Q&A]
Hi all, 

I'm trying to deploy to my local **ZKsync** node the **SimpleStorage.sol** contract, but to no avail.
The command I'm running is the following:

`forge create ./src/SimpleStorage.sol:SimpleStorage --rpc-url http://127.0.0.1:8011 --private-key 0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 --legacy --zksync`

I believe I followed all the instructions: I used all the flags used in the video and pasted in one of the **private keys** that the documentation provides, I entered the correct **rpc-url**, my **docker** is up and i'm using the right forge version with **foundry-zksync**, but it returns this **output**:

> [⠃] Using zksolc-1.5.1
> [⠊] Compiling (zksync)
> [⠒] Compiling 1 files with ZkSolc 0.8.19
> [⠢] ZkSolc 0.8.19 finished in 5.51ms
> Error: 
> Compiler run failed:
> Error
> **Invalid EVM version requested.**

It appears to be using all the **ZkSocl** compilers as the documentation indicates, but it's as if the `--zksync` command didn't fire at all as it doesn't recognize the **zkEVM**.

If anyone has already encountered this error, please let me know if it has been resolved somehow.

**Comment by EngrPips** [comment]
Hello @Alessandro-Cavaliere, did you switch to `foundry-zksync` by running `foundryup-zksync`?

**Reply by Alessandro-Cavaliere** [reply]
Yes, as i written in my question :" i'm using the right forge version with foundry-zksync".

`$ forge --version`
`forge 0.0.2 (4f2145e 2024-07-30T00:22:38.739693553Z)`

**Reply by EngrPips** [reply]
I am not sure what the problem but honestly, I will say you should continue with the course as testing on local `zksync` chain is optional, and Patrick made it clear we should not stress it if things don't work correctly as `Anvil` is enough to test all our process.

**Reply by Alessandro-Cavaliere** [reply]
I know, in fact I'm continuing and leaving the problem aside. I was just curious to know why because running your own node locally is very important. I will contact `ZKsync` directly. Thank you anyway.

**Reply by EngrPips** [reply]
You are welcome; if you contact `ZkSync` and get a solution to the problem, please come back here to post the solution so that it will help any other person who runs into the same problem.

**Comment by maktriss** [comment]
Hi,  @Alessandro-Cavaliere, I did have the same problem, change pragma solidity 0.8.19; to 0.8.25;

**Reply by EngrPips** [reply]
Thanks for the Heads-up

**Reply by Pignatar0md** [reply]
This solved my issue, but I used 0.8.24

**Reply by Alessandro-Cavaliere** [reply]
> Hi, @Alessandro-Cavaliere, I did have the same problem, change pragma solidity 0.8.19; to 0.8.25;

Hello, I tried now: 

`forge create ./src/SimpleStorage.sol:SimpleStorage --private-key 0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356 --rpc-url http://127.0.0.1:8545 --legacy --zksync`             

> [⠃] Using zksolc-1.5.1
> [⠊] Compiling (zksync)
> [⠒] Compiling 1 files with ZkSolc 0.8.25
> [⠑] ZkSolc 0.8.25 finished in 536.59ms
> Compiler run successful!
> Error: 
> **failed rpc call for estimating fee: ErrorResp(ErrorPayload { code: -32601, message: "Method not found", data: None })**

It gives me another error.

**Reply by EngrPips** [reply]
Working with zksync is optional. You can skip the part and continue. I wouldn't let that stop me from moving forward, as `Anvil` is enough for all your tests.

**Reply by Alessandro-Cavaliere** [reply]
> Working with zksync is optional. You can skip the part and continue. I wouldn't let that stop me from moving forward, as `Anvil` is enough for all your tests.

Thank you for the advice, but I'm solving this problem in my spare time, I'm not giving it much importance.

**Reply by chayanbawn** [reply]
Updating solidity compiler version to latest one solved my problem. 

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.26;

```

**Reply by EngrPips** [reply]
Awesome

**Comment by stormcloud266** [comment]
Adding `evm_version = 'istanbul'` to my foundry.toml file fixed this error for me

**Reply by EngrPips** [reply]
That is interesting. Thanks for the heads-up, buddy.

**Reply by AupikZupik** [reply]
Hey It worked. Thanks

**Reply by EngrPips** [reply]
Awesome.

**Comment by Alessandro-Cavaliere** [comment]
I also contacted ZKSync support, you can go and see the response they gave me here: [ZKsync Discussion](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/641)

**Reply by EngrPips** [reply]
Hey friend, Did the bot suggestion help you resolve the issue?

**Reply by Alessandro-Cavaliere** [reply]
> Hey friend, Did the bot suggestion help you resolve the issue?

Unfortunately no, I won't waste any more time on it though, I'll close the discussion.

**Reply by EngrPips** [reply]
Maybe sometime we will find a solution to this.
